src/Items/ItemBanner.h

```diff
@@ -0,0 +1,221 @@
+	{
+		return true;
+	}
+
+
+
+
+
+	virtual bool GetBlocksToPlace(
+		cWorld & a_World,
+		cPlayer & a_Player,
+		const cItem & a_EquippedItem,
+		const Vector3i a_PlacedBlockPos,
+		eBlockFace a_ClickedBlockFace,
+		const Vector3i a_CursorPos,
+		sSetBlockVector & a_BlocksToPlace
+	) override
+	{
+		NIBBLETYPE BlockMeta = 0x00;
+		double Rotation = a_Player.GetYaw();
+		// Placing on the floor
+		if (a_ClickedBlockFace == BLOCK_FACE_TOP)
+		{
+			if ((Rotation >= - 11.25f) && (Rotation < 11.25f))
+			{
+				// South
+				BlockMeta |= 0x08;
+			}
+			else if ((Rotation >= 11.25f) && (Rotation < 33.75f))
+			{
+				// SouthSouthWest
+				BlockMeta |= 0x09;
+			}
+			else if ((Rotation >= 23.75f) && (Rotation < 56.25f))
+			{
+				// SouthWest
+				BlockMeta |= 0x0a;
+			}
+			else if ((Rotation >= 56.25f) && (Rotation < 78.75f))
+			{
+				// WestSouthWest
+				BlockMeta |= 0x0b;
+			}
+			else if ((Rotation >= 78.75f) && (Rotation < 101.25f))
+			{
+				// West
+				BlockMeta |= 0x0c;
+			}
+			else if ((Rotation >= 101.25f) && (Rotation < 123.75f))
+			{
+				// WestNorthWest
+				BlockMeta |= 0x0d;
+			}
+			else if ((Rotation >= 123.75f) && (Rotation < 146.25f))
+			{
+				// NorthWest
+				BlockMeta |= 0x0e;
+			}
+			else if ((Rotation >= 146.25f) && (Rotation < 168.75f))
+			{
+				// NorthNorthWest
+				BlockMeta |= 0x0f;
+			}
+			else if ((Rotation >= -168.75f) && (Rotation < -146.25f))
+			{
+				// NorthNorthEast
+				BlockMeta |= 0x01;
+			}
+			else if ((Rotation >= -146.25f) && (Rotation < -123.75f))
+			{
+				// NorthEast
+				BlockMeta |= 0x02;
+			}
+			else if ((Rotation >= -123.75f) && (Rotation < -101.25f))
+			{
+				// EastNorthEast
+				BlockMeta |= 0x03;
+			}
+			else if ((Rotation >= -101.25) && (Rotation < -78.75f))
+			{
+				// East
+				BlockMeta |= 0x04;
+			}
+			else if ((Rotation >= -78.75) && (Rotation < -56.25f))
+			{
+				// EastSouthEast
+				BlockMeta |= 0x05;
+			}
+			else if ((Rotation >= -56.25f) && (Rotation < -33.75f))
+			{
+				// SouthEast
+				BlockMeta |= 0x06;
+			}
+			else if ((Rotation >= -33.75f) && (Rotation < -11.25f))
+			{
+				// SouthSouthEast
+				BlockMeta |= 0x07;
+			}
+			else  // degrees jumping from 180 to -180
+			{
+				// North
+				BlockMeta |= 0x00;
+			}
+			a_BlocksToPlace.emplace_back(a_PlacedBlockPos, E_BLOCK_STANDING_BANNER, BlockMeta);
+		}
+		// placing on the sides
+		else if (a_ClickedBlockFace != BLOCK_FACE_NONE)
+		{
+			if (a_ClickedBlockFace == BLOCK_FACE_EAST)
+			{
+				BlockMeta |= 0x05;
+			}
+			else if (a_ClickedBlockFace == BLOCK_FACE_WEST)
+			{
+				BlockMeta |= 0x04;
+			}
+			else if (a_ClickedBlockFace == BLOCK_FACE_NORTH)
+			{
+				BlockMeta |= 0x02;
+			}
+			else  // degrees jumping from 180 to -180
+			{
+				BlockMeta |= 0x03;
+			}
+			a_BlocksToPlace.emplace_back(a_PlacedBlockPos, E_BLOCK_WALL_BANNER, BlockMeta);
+		}
+		else
+		{
+			return false;
+		}
+		return true;
+	}
+
+
+
+
+
+	virtual bool OnPlayerPlace(
```

Comment:
I think it's possible to move the blockentity setting to cBlockBannerHandler::OnPlacedByPlayer; this would help reduce duplication

Reply:
No - we have to know the item's metadata to determine the color of the banner which is only known in `OnPlayerPlace`